tests/quic_p2p.rs

```diff
@@ -0,0 +1,40 @@
+use quic_p2p::{Config, Event, QuicP2p};
+use std::net::{IpAddr, Ipv4Addr};
+use std::sync::mpsc;
+use unwrap::unwrap;
+
+#[test]
+fn successfull_connection_stores_peer_in_bootstrap_cache() {
+    let (ev_tx, _ev_rx) = mpsc::channel();
+    let mut peer1 = QuicP2p::with_config(
+        ev_tx,
+        Config {
+            port: Some(0),
+            ip: Some(IpAddr::V4(Ipv4Addr::LOCALHOST)),
+            ..Default::default()
+        },
+    );
+    unwrap!(peer1.start_listening());
+    let peer1_conn_info = unwrap!(peer1.our_connection_info());
+
+    let (ev_tx, ev_rx) = mpsc::channel();
+    let mut peer2 = QuicP2p::with_config(
+        ev_tx,
+        Config {
+            port: Some(0),
+            ip: Some(IpAddr::V4(Ipv4Addr::LOCALHOST)),
+            ..Default::default()
+        },
+    );
+    unwrap!(peer2.start_listening());
+    peer2.connect_to(peer1_conn_info.clone());
+
+    for event in ev_rx.iter() {
+        if let Event::ConnectedTo { .. } = event {
+            break;
+        }
+    }
+
+    let cache = unwrap!(peer2.bootstrap_cache());
+    assert_eq!(cache, vec![peer1_conn_info]);
```

Comment:
Maybe also assert that `peer1` OTOH did also get `ConnectedTo` but still did not add `peer2` in its cache.

Reply:
I'll make this a separate test case - don't want a single test check too many things :)

Reply:
Hmm it's just two lines i guess. Not sure about the verbosity that so many test cases add. By similar logic routing probably would have 1000s of test cases. I think a single test case is far easier to look at as the reader can maintain a flow while evaluating/reading it. Having too many small test cases just to prove what could be done is a few extra lines in the same test case seems overkill to me. I don't know what the best practice is though.

Reply:
Hmm https://github.com/maidsafe/quic-p2p/pull/26/files#r275857033 is still not covered with in this update of the PR. So pls confirm that point (sp. the 2nd part of the sentence which i'm mainly interested in)

Reply:
Good catch, there was actually a bug, fixed it :)